drivers/CoreInterruptIn/include/CoreInterruptIn.h

```diff
@@ -0,0 +1,29 @@
+// Leka - LekaOS
+// Copyright 2023 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "CoreEventQueue.h"
+#include "interface/drivers/InterruptIn.h"
+
+namespace leka {
+
+class CoreInterruptIn : public interface::InterruptIn
+{
+  public:
+	explicit CoreInterruptIn(PinName pin) : _interrupt_in(pin) { _event_queue.dispatch_forever(); };
+
+	auto read() -> int final;
+
+	void rise(std::function<void()> const &callback) final;
+	void fall(std::function<void()> const &callback) final;
+
+  private:
+	CoreEventQueue _event_queue;
```

Comment:
as explained on slack, this should be removed.

interrupts are very low level hardware primitive used to react **instantly** when they trigger.

event queues are used to differ execution of slow parts of code outside of ISR context.

the fact that we need to use an event queue after an interrupt is an implementation detail and should be handled by the developper.

CoreInterruptIn must be kept as low level and fast as possible.